app/models/task.rb

```diff
@@ -1,3 +1,14 @@
 class Task < ActiveRecord::Base
+  before_save :set_completed_field_to_false_for_new_records
+
   validates :name, presence: true
+
+  belongs_to :list
+
+  scope :completed_tasks, -> { where completed: true }
+  scope :incompleted_tasks, -> { where completed: false }
+
+  def set_completed_field_to_false_for_new_records
+    self.completed = false if self.new_record?
```

Comment:
What was you thinking in setting this here, vs. setting a default val for the attr in the migration?

Reply:
[question] Aside from Danny's suggestion, were there any other approaches you considered here as well?

Reply:
My original approach was just setting the default value as `false`. However, because `:completed` field is permitted in the `#task_params`, I thought a user with malicious intent could send the attribute through the params with wrong value. 
Therefore, I decided to have a safeguard to validate on the model level using ActiveRecord's `before_save` and `.new_record?` methods in addition to setting the default value to false.